### PR TITLE
Add cw721-receiver contract

### PR DESCRIPTION
**Pull request by chipshort**
This implements an example contract that can receive a cw721 token and then just succeed or error depending on the passed message.

Also fine if you don't want to merge. We just needed this for a wasmd test. In that case, just close it.

### CODE REVIEW

**Comment by JakeHartnell** [comment]
This is really helpful! Have had to make similar contracts to this myself... maybe we need a `test-contracts` folder? Open to ideas from others but would be down to have some test contracts like this.

**Review by shanev — APPROVED** [review]

**Comment by Art3miX** [comment]
> This is really helpful! Have had to make similar contracts to this myself... maybe we need a `test-contracts` folder? Open to ideas from others but would be down to have some test contracts like this.

I think having a `test-contracts` folder is smart, you almost always need a test contract if you work with cw721

File: contracts/cw721-receiver/Cargo.toml
Comment (Art3miX) [inline contracts/cw721-receiver/Cargo.toml]: Not needed as you don't use them in the contract?

File: contracts/cw721-receiver/Cargo.toml
Comment (Art3miX) [inline contracts/cw721-receiver/Cargo.toml]: Only used in tests, so can move to the dev-deps?

File: contracts/cw721-receiver/tests/multitest.rs
Comment (Art3miX) [inline contracts/cw721-receiver/tests/multitest.rs]: We can remove the print and assert the attributes are what we expect them to be.

I would also add a test for the failed option and for wrong msg, just in case for future changes.

**Review by Art3miX — COMMENTED** [review]
Cute little contract.

P.s, need CI fix.

File: contracts/cw721-receiver/Cargo.toml
Comment (chipshort) [reply contracts/cw721-receiver/Cargo.toml]: I moved cw721-base. cw721 is needed because I reuse the `Cw721ReceiveMsg` in the contract's execute msg.

File: contracts/cw721-receiver/tests/multitest.rs
Comment (chipshort) [reply contracts/cw721-receiver/tests/multitest.rs]: Done 👍

**Comment by chipshort** [comment]
I applied @Art3miX's feedback.
Should I create a `test-contracts` folder and move it there?

**Review by Art3miX — APPROVED** [review]
LGTM

**Comment by taitruong** [comment]
> I applied @Art3miX's feedback. Should I create a `test-contracts` folder and move it there?

Just rename it to `cw721-receiver-tester`, no extra folder needed imo. Once done we can merge it then.

**Comment by chipshort** [comment]
Can someone merge? I don't have write access here

**Review by JakeHartnell — APPROVED** [review]
Amazing! This will be really useful.

**Pull request merged**
